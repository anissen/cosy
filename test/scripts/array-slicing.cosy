print [5,4,2,62,1][2..4] == [2, 62]

{
    mut x = [5,4,2,62,1]
    print x
    print x[2..4]
    x[2..4] = [9]
    print x
    print x[2..4]
}

{
    mut x = [2,3,4,5,6,7]
    print x
    x[1..2] = [6,7,8]
    print x
    x[-1..1] = [6,7,8]
    print x
    x[1..5] = [8,9]
    print x
    x[0..2] += 10
    print x
    print x == [12,18,9]
    print x[1..3]
    print x[3..1]
    print x[-1..1]
}