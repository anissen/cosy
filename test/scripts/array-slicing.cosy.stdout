true
[5,4,2,62,1]
[2,62]
[5,4,9,1]
[9,1]
[2,3,4,5,6,7]
[2,6,7,8,6,7,4,5]
[2,6,7,8,5]
[2,8,9]
[12,18,9]
true
[18,9]
[9,18]
[18]
[e,l,l,o]
[l,l,o]
[h,e,l]
[h,e,l,l]
[h,e,l,l,o]
